test/NuGet.Clients.Tests/NuGet.PackageManagement.VisualStudio.Test/Services/NuGetPackageFileServiceTests.cs

```diff
@@ -112,6 +111,47 @@ public async Task GetPackageIconAsync_EmbeddedFromFallbackFolder_HasOnlyReadAcce
             }
         }
 
+        [Fact]
+        public async Task GetPackageIconAsync_RemoteUriInaccessible_FindsLocalIconFromCache()
+        {
+            // Arrange
+            using (TestDirectory testDirectory = TestDirectory.Create())
+            {
+                string fileContents = "testFileContents";
+                string filePath = Path.Combine(testDirectory.Path, "testFile.txt");
+
+                File.WriteAllText(filePath, fileContents);
+
+                var packageFileService = new NuGetPackageFileService(
+                        default(ServiceActivationOptions),
+                        Mock.Of<IServiceBroker>(),
+                        new AuthorizationServiceClient(Mock.Of<IAuthorizationService>()),
+                        _telemetryProvider.Object);
+
+                var packageIdentity = new PackageIdentity("packageA", NuGetVersion.Parse("1.0.0"));
+                var remoteInaccessibleUri = new Uri("http://api.source/index.json");
```

Comment:
Sure.. I wondered about that but as you may guess, I found this in many other tests. I've never heard of `.source` so figured we're using it because it's safe. I'll switch this one but I won't be refactoring all tests in this PR (meaning your concern will still exist in this repo)